schemes/rasch_kristjansson/rk_stratiform.F90

```diff
@@ -0,0 +1,571 @@
+  public :: rk_stratiform_cloud_fraction_perturbation_run         ! see note.
+  public :: rk_stratiform_external_forcings_run
+  public :: rk_stratiform_condensate_repartioning_run
+  ! -- prognostic_cloud_water --
+  public :: rk_stratiform_prognostic_cloud_water_tendencies_run
+  public :: rk_stratiform_cloud_optical_properties_run
+  public :: rk_stratiform_save_qtlcwat_run
+
+  !
+
+contains
+
+  ! Initialize rk_stratiform
+!> \section arg_table_rk_stratiform_init Argument Table
+!! \htmlinclude arg_table_rk_stratiform_init.html
+  subroutine rk_stratiform_init( &
+    errmsg, errflg)
+
+    ! Output arguments
+    character(len=512), intent(out)   :: errmsg         ! error message
+    integer,            intent(out)   :: errflg         ! error flag
+
+    errmsg = ''
+    errflg = 0
+
+  end subroutine rk_stratiform_init
+
+!> \section arg_table_rk_stratiform_check_qtlcwat_run Argument Table
+!! \htmlinclude arg_table_rk_stratiform_check_qtlcwat_run.html
+  subroutine rk_stratiform_check_qtlcwat_run( &
+    ncol, pver, &
+    t, q_wv, cldice, cldliq, &
+    qcwat, tcwat, lcwat, &  ! from end of last microphysics/macrophysics call.
+    errmsg, errflg)
+
+    ! Input arguments
+    integer,            intent(in)    :: ncol
+    integer,            intent(in)    :: pver
+    real(kind_phys),    intent(in)    :: t(:,:)         ! air_temperature [K]
+    real(kind_phys),    intent(in)    :: q_wv(:, :)     ! adv: water_vapor_mixing_ratio_wrt_moist_air_and_condensed_water [kg kg-1]
+    real(kind_phys),    intent(in)    :: cldice(:,:)    ! adv: cloud_ice_mixing_ratio_wrt_moist_air_and_condensed_water [kg kg-1]
+    real(kind_phys),    intent(in)    :: cldliq(:,:)    ! adv: cloud_liquid_water_mixing_ratio_wrt_moist_air_and_condensed_water [kg kg-1]
+
+    ! Input/output arguments
+    real(kind_phys),    intent(inout) :: qcwat(:,:)     ! [kg kg-1]
+    real(kind_phys),    intent(inout) :: tcwat(:,:)     ! [K]
+    real(kind_phys),    intent(inout) :: lcwat(:,:)     ! [kg kg-1]
+
+    ! Output arguments
+    character(len=512), intent(out)   :: errmsg         ! error message
+    integer,            intent(out)   :: errflg         ! error flag
+
+    errmsg = ''
+    errflg = 0
+
+    ! Check that qcwat and tcwat were initialized - if not then initialize
+    ! this is made as a separate "run" scheme so it does not have to be used in current CAM
+
+    ! The registry will set default values of negative if not read from snapshot.
+    if(qcwat(1,1) < 0._kind_phys) then
+      qcwat(:ncol,:) = q_wv(:ncol,:)
+    endif
+
+    if(tcwat(1,1) < 0._kind_phys) then
+      tcwat(:ncol,:) = t(:ncol,:)
+    endif
+
+    ! lcwat is initialized from initial conditions of cldice, cldliq
+    ! TODO: check if this is always done (appears to be from physpkg.F90) or should be read from snapshot.
+    if(qcwat(1,1) < 0._kind_phys .or. tcwat(1,1) < 0._kind_phys) then
+      lcwat(:ncol,:) = cldice(:ncol,:) + cldliq(:ncol,:)
+    endif
```

Comment:
Looking at the logic above won't this if-statement always be false assuming `q_wv` and `t` have valid values?